cypress/integration/form.spec.ts

```diff
@@ -0,0 +1,70 @@
+// const book_title = 'The Little Schemer';
+
+describe('Form Actions', () => {
+  beforeEach(() => {
+    cy.visit('/#/examples/form');
+  });
+
+  it('should be able to input all values', () => {
+    cy.get('mat-slide-toggle[formcontrolname="autosave"]').click();
+    cy.get('mat-form-field input[formcontrolname="username"]').type(
+      'someusername'
+    );
+    cy.get('input[formcontrolname="password"]').type('somepassword');
+    cy.get('input[formcontrolname="email"]').type('some@email.com');
+    cy.get('input[formcontrolname="birthday"]').type('1/1/2000');
+    cy.get('textarea[formcontrolname="description"]').type('some description');
+    cy.get('mat-checkbox[formcontrolname="requestGift"] label').click();
+
+    // TODO: slider
+    cy.get('mat-slider[formcontrolname="rating"]').click();
+  });
+
+  it('should show validation messages', () => {
+    // no validation errors with pristine form
+    cy.get('mat-error').should('have.length', 0);
+
+    // required validation messages appear when click "Send"
+    cy.contains('button', 'Send').click();
+    cy.get('mat-error').should('have.length', 5);
+    cy.get('form')
+      .should('contain', 'Username is required')
+      .should('contain', 'Password is required')
+      .should('contain', 'E-mail is required')
+      .should('contain', 'Birthday is required')
+      .should('contain', 'Description is required');
+
+    // email format
+    cy.get('form input[formcontrolname="email"]').type('a');
```

Comment:
I'm not sure if we should be this detailed on form errors, because the validators should already be covered by Angular and/or in our unit tests.

We can leave it here tho as an example, just wanted to point this out.
Thoughts @tomastrajan ?

Reply:
seems useful, and we can always delete it if it causes problems?

Reply:
You're right